### ISSUE TITLE
get_overlapping_bodies() in Area2D, always returns an empty array

### ISSUE THREAD

**Issue by OlexiyKravchuk** [godotengine/godot#29562]
Godot_v3.1.1-stable_win64.exe
Windows 7

get_overlapping_bodies() in Area2D, returns an empty array under any circumstances.

**Comment by KoBeWi** [comment]
Reproduction project? I just tested it and it works fine.

**Comment by avencherus** [comment]
I've been using it fine as well.  There is a case I think where turning masks on and off will have a frame of delay before detection happens, and sometimes not until the bodies move.

So there might be some circumstance where it isn't working perfectly, but I haven't quite narrowed down the conditions.

@OlexiyKravchuk If you can document or demonstrate the circumstances it would be appreciated.

**Comment by OlexiyKravchuk** [comment]
@KoBeWi @avencherus
Forgive me, it was probably a false alarm, I could not find out what the reason was and why physics refused to work properly.
I have deadlines, and I have been spending on this problem for 3 days already and I can no longer look for reasons, so I just redid everything in a new way, and now everything seems to be working.
I now have only one guess, why it did not work in the first version ...
To realize the explosions that caused damage to the area at a defined point in time, I used the method ($"Explosion/CollisionShape2D".set_deferred("disabled", false)) to include a detection zone, which was to take all the bodies in the radius of the lesion and call into them the function of dealing damage.
Now I did everything the other way, avoiding this function.
My guess was that the physics server did not have time to process the overlap of bodies in this frame (in which the function included the form), and waited for the next frame, but then it would be too late to process the event, since by the time of the next frame the object should be destroyed both from one and from the other side of the interaction.
Based on this, I close this problem.
